vision/feed_handler.py

```diff
@@ -0,0 +1,83 @@
+import cv2
+import time
+import sys
+import multiprocessing as mp
+
+# Pyfakewebcam requires linux
+if sys.platform == "linux":
+    import pyfakewebcam
+
+
+class FeedHandler:
+    def __init__(self, resolution_x=640, resolution_y=480, frame_rate=30):
+        '''
+        Configure the resolution and framerate of all feed handlers
+        '''
+
+        self.feeds = {}
+        self.fourcc = cv2.VideoWriter_fourcc(*'XVID')
+        self.resolution_x = resolution_x
+        self.resolution_y = resolution_y
+        self.frame_rate = frame_rate
+
+    def feed_process(self, pipe, num, feed_id, save_video=True, stream_video=True):
+        '''
+        Function to be run as a process, configures streaming and recording of frames.
+        Uses pipe to receive incoming frames and stream/save
+        '''
+
+        # Only attempt to stream video if on Linux (due to package dependancies)
+        if stream_video and sys.platform == "linux":
+            streamer = pyfakewebcam.FakeWebcam(f'/dev/video{num}', self.resolution_x, self.resolution_y)  # append v4l output to list of cameras
+
+        if save_video:
+            video_filename = f'logs/stream_{feed_id}_' + time.strftime("%Y%m%d-%H%M%S")  # save videos to unique files
+            video_writer = cv2.VideoWriter(video_filename + "_left.avi", self.fourcc, self.frame_rate, (self.resolution_x, self.resolution_y))  # append video writer to list of video writers
+
+        p_output, p_input = pipe
+        p_input.close()    # We are only reading
+
+        while True:
+            data = p_output.recv()
+            # Resize image to reduce bandwidth/size
+            image = cv2.resize(data, (640, 480))
```

Comment:
I would consider making this resolution either constants defined somewhere or parameters passed in so that we aren't obscuring how to get a higher resolution image. Currently, if I passed a different resolution into the constructor, I would expect that resolution to be streamed and saved. If not, why support custom resolutions at all?